src/modules/nim.rs

```diff
@@ -0,0 +1,124 @@
+use super::{Context, Module, RootModuleConfig, SegmentConfig};
+
+use crate::configs::nim::NimConfig;
+use crate::utils;
+
+/// Creates a module with the current Nim version
+///
+/// Will display the Nim version if any of the following criteria are met:
+///     - The current directory contains a file with extension `.nim`, `.nims`, or `.nimble`
+///     - The current directory contains a `nim.cfg` file
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let is_nim_project = context
+        .try_begin_scan()?
+        .set_files(&["nim.cfg"])
+        .set_extensions(&["nim", "nims", "nimble"])
+        .is_match();
+
+    if !is_nim_project {
+        return None;
+    }
```

Comment:
Pad with a blank line: 
```suggestion
    }

```